core/src/test/java/org/elasticsearch/cluster/routing/RoutingTableTests.java

```diff
@@ -297,31 +297,54 @@ public void testValidations() {
                                                    .numberOfShards(numShards)
                                                    .numberOfReplicas(numReplicas)
                                                    .build();
+        MetaData metaData = MetaData.builder().put(indexMetaData, true).build();
         final RoutingTableGenerator routingTableGenerator = new RoutingTableGenerator();
         final RoutingTableGenerator.ShardCounter counter = new RoutingTableGenerator.ShardCounter();
         final IndexRoutingTable indexRoutingTable = routingTableGenerator.genIndexRoutingTable(indexMetaData, counter);
         // test no validation errors
-        assertThat(indexRoutingTable.validate(indexMetaData), Matchers.hasSize(0));
+        try {
+            indexRoutingTable.validate(metaData);
+        } catch (IllegalStateException e) {
```

Comment:
nit: you can just let the validate throw it's exception - this will fail the test and give us more info